relay-filter/src/generic_filters.rs

```diff
@@ -0,0 +1,106 @@
+//! Implements generic filtering based on the [`RuleCondition`] DSL.
+//!
+//! Multiple generic filters can be defined and they are going to be checked in FIFO order. The
+//! first one that matches, will result in the event being discarded with a [`FilterStatKey`]
+//! identifying the matching filter.
+
+use crate::{FilterStatKey, GenericFiltersConfig};
+use relay_event_schema::protocol::Event;
+use relay_protocol::RuleCondition;
+
+/// Version of the generic filters schema.
+pub const VERSION: u16 = 1;
+
+/// Checks events by patterns in their error messages.
+pub fn matches(event: &Event, condition: Option<&RuleCondition>) -> bool {
+    // TODO: the condition DSL needs to be extended to support more complex semantics, such as
+    //  collections operations.
+    condition.map_or(false, |condition| condition.matches(event))
+}
+
+/// Filters events by patterns in their error messages.
+pub fn should_filter(event: &Event, config: &GenericFiltersConfig) -> Result<(), FilterStatKey> {
```

Comment:
I'd move the maximum version check here. It'd also be great if we could test it.